content/blog/22-23/reverse_engineering_epaper.md

```diff
@@ -0,0 +1,141 @@
+> to make an omelet; likewise, you have to be willing to sacrifice
+> devices to hack a system. Fortunately, acquiring multiple
+> copies of a mass-produced piece of hardware is easy. I often
+> do a bit of dumpster diving or check classified advertisements
+> to get sample units for research purposes. I generally try to
+> start with three copies: one to tear apart and never put back
+> together, one to probe, and one to keep relatively pristine.
+
+After gently prying apart the case, we got a closer look at the printed circuit board. This PCB had the e-ink screen and a battery connected to it. The battery was immediately disconnected before further work was done (as a safety precaution, to not accidentally short something). Pictures were then taken of the front and back of the PCB, see picture below for an explanation of the components.
+
+<%= figure 'https://pics.zeus.gent/9aUXMspqgZdJ2jzKalyo2bJ2nWsUWTh0ZTd2GPCS.jpg', 'Front of the PCB' %>
+
+<%= figure 'https://pics.zeus.gent/2RNNRTsWSAAFV4Kf5cLsTJAazGi02fzA2CX8mifP.jpg', 'Back of the PCB' %>
+
+
+The product is an SES-imagotag G1 2.7 BW NFC, with product code B27N02003. On the PCB, there is some text: RFRTx002D and KIM 1514 (most likely internal part numbers), along with (most likely) a datecode: 09 17 (so September 2017). The PCB itself looks very professionally designed and made, with plenty of testpoints and programmer pads. Attached to the screen, there is an NFC tag which contains the ID of the board.
+
+The board contains a CC2510 microcontroller. This microcontroller can communicate over 2.4GHz, and it's clear that the product uses this feature: there is an antenna structure present on the PCB. It's very likely that the image updates are done wirelessly via that antenna. The only remaining part is then to figure what the communication protocol looks like. Unfortunately, we don't have the accompanying device that puts images on the screen, so intercepting the wireless communications wouldn't be possible. What would be possible however, is reading the code from the microcontroller and seeing what it expects.
+
+The pictures of the PCB were transformed and overlaid in GIMP. The front and back layers of the PCB were put in different GIMP layers, so that it would be easy to switch between different views without losing your frame of reference. When routing signals between places of the board, PCB designers sometimes have to use so-called 'vias' to move from the front of the board to the back of the board. From the marking '1 TOP' and '4 BOT' on the PCB, it became clear that this is 4-layer PCB: there are two visible layers, and two hidden layers sandwitched between them. Luckily, someone else used acid and sandpaper to [make the hidden layers visible](https://epongenoir.blogspot.com/2017/07/reverse-engineering-imagotag-retail-26.html). This was however on a slightly different revision, but most components are in roughly the same position.
```

Comment:
```suggestion
The pictures of the PCB were transformed and overlaid in GIMP. The front and back layers of the PCB were put in different GIMP layers, so that it would be easy to switch between different views without losing your frame of reference. When routing signals between places of the board, PCB designers sometimes have to use so-called 'vias' to move from the front of the board to the back of the board. From the marking '1 TOP' and '4 BOT' on the PCB, it became clear that this is 4-layer PCB: there are two visible layers, and two hidden layers sandwiched between them. Luckily, someone else used acid and sandpaper to [make the hidden layers visible](https://epongenoir.blogspot.com/2017/07/reverse-engineering-imagotag-retail-26.html). This was however on a slightly different revision, but most components are in roughly the same position.
```